internal/runbits/buildscript/merge.go

```diff
@@ -0,0 +1,58 @@
+package buildscript
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/pkg/localcommit"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildexpression/merge"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildscript"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+// Merge merges the local build script with the remote buildexpression (not script) for a given
+// UUID, performing the given merge strategy (e.g. from model.MergeCommit).
+func Merge(proj *project.Project, remoteCommit strfmt.UUID, strategies *mono_models.MergeStrategies, auth *authentication.Auth) error {
+	// Verify we have a build script to merge.
+	script, err := buildscript.NewScriptFromProjectDir(proj.Dir())
+	if err != nil && !buildscript.IsDoesNotExistError(err) {
+		return errs.Wrap(err, "Could not get local build script")
+	}
+	if script == nil {
+		return nil // no build script to merge
+	}
+
+	// Get the local and remote build expressions to merge.
+	bp := model.NewBuildPlannerModel(auth)
+	localCommit, err := localcommit.Get(proj.Dir())
+	if err != nil {
+		return errs.Wrap(err, "Unable to get local commit")
+	}
+	exprA, err := bp.GetBuildExpression(proj.Owner(), proj.Name(), localCommit.String())
+	if err != nil {
+		return errs.Wrap(err, "Unable to get buildexpression for local commit")
+	}
```

Comment:
The reason is that the build expression might contain data that is not in the build script. I don't know exactly what that is, but I wanted to err on the side of caution here.

Reply:
Actually that just convinced me that we should use the local buildscript instead. If there are differences then it will lead to a conflict, and the user can resolve those differences.

With the current implementation, it seems that if we encounter this scenario then we would silently blow away the users customizations.

As you say it shouldn't happen, but if we're going to err on the side of safety then it feels more safe to use the local buildscript.